tests/integration/test_agentd/test_agentd_state_config/test_agentd_state_config.py

```diff
@@ -0,0 +1,157 @@
+
+# Marks
+pytestmark = pytest.mark.tier(level=0)
+
+# Configuration and cases data.
+configs_path = Path(CONFIGS_PATH, 'wazuh_conf.yaml')
+cases_path = Path(TEST_CASES_PATH, 'wazuh_state_config_tests.yaml')
+
+# Test configurations.
+config_parameters, test_metadata, test_cases_ids = get_test_cases_data(cases_path)
+
+test_configuration = load_configuration_template(configs_path, config_parameters, test_metadata)
+
+if sys.platform == WINDOWS:
+    local_internal_options = {AGENTD_WINDOWS_DEBUG: '2'}
+else:
+    local_internal_options = {AGENTD_DEBUG: '2'}
+
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_agentd_state_config(test_configuration, test_metadata, remove_state_file, set_wazuh_configuration, configure_local_internal_options,
+                             truncate_monitored_files):
+    
+    '''
+    description: Check that the 'wazuh-agentd.state' statistics file is created
+                 automatically and verify that it is updated at the set intervals.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: data
+            brief: Configuration used in the test.
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - remove_state_file:
+            type: fixture
+            brief: Removes wazuh-agentd.state file.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+
+    assertions:
+        - Verify that the 'wazuh-agentd.state' statistics file has been created.
+        - Verify that the 'wazuh-agentd.state' statistics file is updated at the specified intervals.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Different test cases that are contained in an external YAML file (wazuh_state_config_tests.yaml)
+                       that includes the parameters and their expected responses.
+
+    expected_output:
+        - '.*Invalid definition for agent.state_interval.*'
+        - '.*State file is disabled.*'
+        - '.*State file updating thread started.*'
+    '''
+
+    control_service('stop', 'wazuh-agentd')
+
+    if sys.platform == WINDOWS:
+        if test_metadata['agentd_ends']:
+            with pytest.raises(ValueError):
+                control_service('start')
+            assert (test_metadata['agentd_ends']
+                    is not check_if_process_is_running('wazuh-agentd'))
+        else:
+            control_service('start')
+    else:
+        control_service('start', 'wazuh-agentd')
```

Comment:
Use framework constants. Fix here and where needed.

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/18353/commits/ecf23d03eafffc97b67a2d364324e916eb50d143